pages/api/generate.ts

```diff
@@ -79,7 +91,9 @@ export default async function handler(
       await new Promise((resolve) => setTimeout(resolve, 1000));
     }
   }
+  const status = endpointResponse(restoredImage, {pass: 200, fail: 500})
+  const json = endpointResponse(restoredImage, {pass: restoredImage, fail: "Failed to restore image"})
   res
-    .status(200)
```

Comment:
The client checks the status of the request before rendering the restored image, when the JSON message is "Failed to restore image" while the status is 200, the client tries to render an image with the url of "Failed%20to%20restore%20image" returning a different status code fixes the issue from #18